libraries/chain/include/eosio/chain/block_header_state.hpp

```diff
@@ -121,12 +129,22 @@ struct block_header_state : fc::reflect_init {
    // When the block associated with a proposed finalizer policy becomes final,
    // it becomes pending.
    std::vector<std::pair<block_num_type, finalizer_policy_ptr>> proposed_finalizer_policies;
+
    // Track in-flight pending finalizer policy. At most one pending
    // finalizer policy at any moment.
    // When the block associated with the pending finalizer policy becomes final,
    // it becomes active.
    std::optional<std::pair<block_num_type, finalizer_policy_ptr>> pending_finalizer_policy;
 
+   // It may be that the `finality_core` references a finalizer policy generation which is neither the active
+   // or pending one. This can happen when a pending policy became active, replacing the previously active
```

Comment:
To be clear, it is possible for the latest final block (which also is tracked in  `finality_core` to have an active finalizer policy that is still not being tracked.

But we don't care about that.

We just care about any finalizer policies tracked within `finality_core` that are restricted to the block range from `core.latest_qc_claim().block_num` to `core.current_block_num()` inclusive.

Reply:
https://github.com/AntelopeIO/spring/pull/719/commits/0dd7a3639f78f350fae5bcf7ecbaa20972337226